### ISSUE TITLE
{General} Falling from a mount / Instant death

### ISSUE THREAD

**Issue by durrrr** [TrinityCore/TrinityCore#6356]
Hello,
I have encountered a problem while playing in any of the game zones.

How to reproduce the bug:
1. Fall down from a mount to the ground (No matter from what distance)

What should happen, what happens instead:
You should dismiss your mount in mid-air and get the correct amount of damage regarding your fall and distance. Instead, sometimes when you dismiss your mount in mid-air situations, no matter what distance from the ground you are, you get a 1 shot, like you fell from the sky. This also happens with a mount, just touching the ground sometimes, after you flew for a long time.

FYI, I can guess it is something with the distance of flight or something, but i have no actual idea  what it is connected with and what links can a provide about it.

I'm running Unix on hash: bf7a32b, TDB 335.11.46+
Addons: AH bot and WG.

Regards,
durrrr

**Comment by Aokromes** [comment]
Cannot reproduce since a lot of time ago.

**Comment by Killyana** [comment]
This was easy to reproduce in icc entrance when the player is dismounted automatically, It was guarantied to be OS, but on rev. 5b5c0938da0f impossible to reproduce.

**Comment by Nyeriah** [comment]
Note: you can't reproduce this locally as it's mostly a result of latency.

**Comment by adeutscher** [comment]
I was able to reproduce this on my local machine in my local branch that broke away at  b91bed0.

After bringing my branch up to date, it looks like this was fixed for good with fcbbc60. If a player uses a mount item or flight form (whether to fly or di the server will go through `AuraEffect::HandleAuraModIncreaseFlightSpeed`. If the right mask is set and the caster is either applying flight or has a flying buff (basically any time barring shenanigans), a player will go into `Player::SetCanFly`. If they are removing flight, then the player will have their fall information reset.

fcbbc60 was written for #16955 with flight form in mind, but it seems to squash hiccups of this sort in general.
